### PR TITLE
KEYCLOAK-9551 Client Credentials Grant should not generate refresh token

### PR DESCRIPTION
**Pull request by mposolda**

### CODE REVIEW

**Comment by mposolda** [comment]
Codebase PR: https://github.com/keycloak/keycloak/pull/7569

**Review by pedroigor — DISMISSED** [review]

File: server_admin/topics/clients/oidc/service-accounts.adoc [hunk 1]
```diff
@@ -45,13 +45,13 @@ Pragma: no-cache
...
-    "refresh_expires_in":600,
-    "id_token":"tGzv3JOkF0XG5Qx2TlKWIA",
-    "not-before-policy":0,
-    "session_state":"234234-234234-234234"
+    "expires_in":60
 }
 ----
 
-The retrieved access token can be refreshed or logged out by an out-of-bound request.
+There is only access token returned by default. There is no refresh token returned and there is also no user session created
```
Comment (andymunro) [inline server_admin/topics/clients/oidc/service-accounts.adoc]: add "the" before "only"

File: server_admin/topics/clients/oidc/service-accounts.adoc [hunk 2]
```diff
@@ -45,13 +45,13 @@ Pragma: no-cache
...
+on {project_name} side upon successful authentication by default. Due the lack of refresh token, there is a need to re-authenticate when access token expires,
```
Comment (andymunro) [inline server_admin/topics/clients/oidc/service-accounts.adoc]: add "the" before "{project_name}"

File: upgrading/topics/keycloak/changes.adoc [hunk 3]
```diff
@@ -14,6 +14,19 @@ However, there are some changes in login theme design, so please upgrade your cu
 There is an example with the changes in `keycloak/examples/themes/theme/sunrise` folder.
 No additional setup is required.
 
+==== Client Credentials Grant without refresh token by default
+
+From this {project_name} version, the OAuth2 Client Credentials Grant endpoint does not issue refresh tokens by default. This behaviour is aligned with the OAuth2 specification.
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes.adoc]: we use the american English spelling "behavior"

File: upgrading/topics/keycloak/changes.adoc [hunk 4]
```diff
@@ -14,6 +14,19 @@ However, there are some changes in login theme design, so please upgrade your cu
...
+As a side-effect of this change, there is no user session created on the {project_name} server side after successful Client Credentials authentication, which results
+in improved performance and memory consumption. Clients, which use Client Credentials Grant are encouraged to stop using
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes.adoc]: no comma after Clients  and it should be "Clients that use...

File: upgrading/topics/keycloak/changes.adoc [hunk 5]
```diff
@@ -14,6 +14,19 @@ However, there are some changes in login theme design, so please upgrade your cu
...
+refresh tokens and instead always authenticate at every request with `grant_type=client_credentials` instead of using `refresh_token` as grant type.
+In relation to this, {project_name} has support for revocation of access tokens in the OAuth2 Revocation Endpoint, hence clients are allowed
+to revoke individual access tokens if needed.
+
+For the backwards compatibility, there is a possibility to stick to the behaviour of old versions. When this is used, refresh token will be still issued after
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes.adoc]: behavior

Comment (andymunro) [inline upgrading/topics/keycloak/changes.adoc]: Add "the" before "refresh token"

File: upgrading/topics/keycloak/changes.adoc [hunk 6]
```diff
@@ -14,6 +14,19 @@ However, there are some changes in login theme design, so please upgrade your cu
...
+successful authentication with Client Credentials Grant and also the user session will be created. This can be enabled for the particular client in
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes.adoc]: "a" successful
and "the" before "Client Credentials  Grant"

**Review by andymunro — CHANGES_REQUESTED** [review]
Approved with a few comments. I will be out on PTO Friday.

**Comment by mposolda** [comment]
@andymunro I've applied all your recommendations. Thanks for the review!

**Pull request merged**